### PR TITLE
[JENKINS-65800] Option to disable only fingerprint produced files or only archive

### PR DESCRIPTION
**Pull request by MatthiasSchmalz**
Add flags to GeneratedArtifactsPublisher to control performed actions, e.g. to only fingerprint but not archive.
See https://issues.jenkins.io/browse/JENKINS-65800

- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your master branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- [NA ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Comment by MatthiasSchmalz** [comment]
I wanted to add a testcase but the whole reporter is not tested and I do not know how to safely Mock all dependencies.

**Comment by MatthiasSchmalz** [comment]
Hi @cyrille-leclerc
could you review this change? It seems like you are the maintainer of this featrue.
Best regards
Matthias

**Comment by aheritier** [comment]
Hi @MatthiasSchmalz 

Thanks a lot for your very valuable contribution. 
@cyrille-leclerc stepped down from the maintenance of this plugin and I am doing my best to follow but I have a very limited time sadly.

Your PR looks globally good to me but there are few things that I would like to review before merging it:
* I think we need to update the documentation/ README to describe the new options
* I have some doubts about the behavior of the new options and the potential conflicts they might have with the default  `disabled` and the marker `.skip-archive-generated-artifacts`. I am afraid about some users confused by the fact that `disabled=true` wins over `archiveFiles=true` and `fingerprintFiles=false`
* For a better coherency, could it be better to have the options reverted and call something like `archiveFilesDisabled` and `fingerprintFilesDisabled` ? WDYT ?
* Another option we could consider is to split the `GeneratedArtifactsPublisher` in two publishers but in such case we have to keep the compatibility with the old one settings which isn't necessarily easy
* In term of tests we have mainly https://github.com/jenkinsci/pipeline-maven-plugin/blob/master/jenkins-plugin/src/test/java/org/jenkinsci/plugins/pipeline/maven/publishers/GeneratedArtifactsReporterTest.java but I agree that such changes aren't easy to test

any feedback @bguerin  ?

**Comment by bguerin** [comment]
Love this PR, makes so much sense.

> update the documentation/ README

agreed, it's a blocker from my point of view

> unit tests

yes, this plugin is a pain from unit tests point of view. It definitely need a work on them

> options naming, backward compatibility, split

I think this PR could introduce 2 news publishers : one for archiving, the other for fingerprinting, and alter the actual big one to mark it deprecated and make it call the 2 others, and by default, the plugin should launch the 2 new.
I think this will keep compatibility, encourage users to switch from the old big publisher to the 2 new, and let the actual behaviour of marker files and function names in pipelines works

**Comment by MatthiasSchmalz** [comment]
@aheritier & @bguerin 
Thank you for the feedback.
I can rename the parameters as proposed. Splitting it makes sense, but I'm not sure if I can manage this.
It might be a future goeal.

File: README.adoc
Comment (bguerin) [inline README.adoc:282]: you target the wrong publisher, changes must be on line 256

**Review by bguerin — CHANGES_REQUESTED** [review]

File: README.adoc
Comment (MatthiasSchmalz) [reply README.adoc]: Thank you for the hint. I have corrected it.

**Comment by bguerin** [comment]
It's OK for me

I'd love to find some time to work on unit tests, so this PR can have some changes on them, to confirm the feature ...
And I also agree that splitting the publisher in two can be done in another PR

@aheritier up to you :)

**Comment by MatthiasSchmalz** [comment]
@aheritier do you have any concerns or could you merge it?

**Comment by aheritier** [comment]
Hi @MatthiasSchmalz 

I think it's fine for me but I didn't have the time to test.
Sadly I will be off and really want to stay AFK in July thus I don't think I will be able to review/merge release before august.
I reiterate my proposal to @bguerin if he wants to co-maintain this plugin, it will be with pleasure ;-)

**Comment by aheritier** [comment]
I created https://issues.jenkins.io/browse/JENKINS-66609 as a followup to remember to split/deprecate it

**Review by aheritier — APPROVED** [review]
👍🏻

**Comment by bguerin** [comment]
@aheritier 

> I reiterate my proposal to @bguerin if he wants to co-maintain this plugin, it will be with pleasure ;-)

Sorry, did not see your message, nor the first one it seems ...
I would love to help maintaining this plugin !

**Comment by aheritier** [comment]
@bguerin can you check : https://github.com/jenkins-infra/repository-permissions-updater/pull/2082 ?

**Comment by bguerin** [comment]
> @bguerin can you check : [jenkins-infra/repository-permissions-updater#2082](https://github.com/jenkins-infra/repository-permissions-updater/pull/2082) ?

Done :)

**Comment by bguerin** [comment]
@aheritier sorry for being silent so long. Permissions on infra are now merged, I would like to begin some things on the plugin this week. I think starting with dependabot updates could be the right way, but it seems I have no rights on these pull requests. Is it the expected behaviour ? Sorry to bother you, it is my first time as a jenkins plugin maintainer :)

**Comment by aheritier** [comment]
it's not expected @bguerin 
It's strange, you are not in https://github.com/orgs/jenkinsci/teams/pipeline-maven-plugin-developers
I will give you the permission manually but it might be great to ask to the jenkins infra if there is not a problem

**Pull request merged**
